test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
     end
 end)
 
+-- Parse string with a custom format, then format datetime object
+-- to a string with the same format and compare initila and
+-- resulted strings:
+--
+--   dt = datetime.parse(<str>, { format = <fmt> })
+--   dt:format(<fmt>) == <str>
+--
+test:test("Parse strptime format and check with :format()", function(test)
+    local formats = {
+        -- %a
```

Comment:
It would be nice to see the definition of field descriptors from the `strptime()` manual for each such descriptor. It is more convenient to read the test without scrolling the manual every time.
Here and below.

Reply:
Added comments.